src/java_binding/src/lib.rs

```diff
@@ -539,13 +540,11 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetTimestampV
     idx: jint,
 ) -> JObject<'a> {
     execute_and_catch(env, move || {
-        let millis = pointer
-            .as_ref()
-            .datum_at(idx as usize)
-            .unwrap()
-            .into_timestamp()
-            .0
-            .timestamp_millis();
+        let scalar_value = pointer.as_ref().datum_at(idx as usize).unwrap();
+        let millis = match scalar_value {
+            ScalarRefImpl::Int64(v) => v / 1000,
```

Comment:
Here assumes the `Timestampz` is in the micro-second unit, we should have some assertions for it.